cmd/nitro/init.go

```diff
@@ -835,106 +813,281 @@ func openInitializeExecutionDB(ctx context.Context, stack *node.Node, config *No
 			}
 			log.Warn("Re-creating genesis though it seems to exist in database", "blockNr", genesisBlockNr)
 		}
 		log.Info("Initializing", "ancients", ancients, "genesisBlockNr", genesisBlockNr)
 		if config.Init.ThenQuit {
 			cacheConfig.SnapshotWait = true
 		}
-		var parsedInitMessage *arbostypes.ParsedInitMessage
-		if config.Node.ParentChainReader.Enable {
-			delayedBridge, err := arbnode.NewDelayedBridge(l1Client, rollupAddrs.Bridge, rollupAddrs.DeployedAt)
-			if err != nil {
-				return executionDB, nil, fmt.Errorf("failed creating delayed bridge while attempting to get serialized chain config from init message: %w", err)
-			}
-			deployedAt := new(big.Int).SetUint64(rollupAddrs.DeployedAt)
-			delayedMessages, err := delayedBridge.LookupMessagesInRange(ctx, deployedAt, deployedAt, nil)
-			if err != nil {
-				return executionDB, nil, fmt.Errorf("failed getting delayed messages while attempting to get serialized chain config from init message: %w", err)
-			}
-			var initMessage *arbostypes.L1IncomingMessage
-			for _, msg := range delayedMessages {
-				if msg.Message.Header.Kind == arbostypes.L1MessageType_Initialize {
-					initMessage = msg.Message
-					break
-				}
-			}
-			if initMessage == nil {
-				return executionDB, nil, fmt.Errorf("failed to get init message while attempting to get serialized chain config")
-			}
-			parsedInitMessage, err = initMessage.ParseInitMessage()
-			if err != nil {
-				return executionDB, nil, err
-			}
-			if parsedInitMessage.ChainId.Cmp(chainId) != 0 {
-				return executionDB, nil, fmt.Errorf("expected L2 chain ID %v but read L2 chain ID %v from init message in L1 inbox", chainId, parsedInitMessage.ChainId)
-			}
-			if parsedInitMessage.ChainConfig != nil {
-				if err := parsedInitMessage.ChainConfig.CheckCompatible(chainConfig, chainConfig.ArbitrumChainParams.GenesisBlockNum, 0); err != nil {
-					return executionDB, nil, fmt.Errorf("incompatible chain config read from init message in L1 inbox: %w", err)
-				}
-			}
-			log.Info("Read serialized chain config from init message", "json", string(parsedInitMessage.SerializedChainConfig))
-		} else {
-			serializedChainConfig, err := json.Marshal(chainConfig)
-			if err != nil {
-				return executionDB, nil, err
-			}
-			parsedInitMessage = &arbostypes.ParsedInitMessage{
-				ChainId:               chainConfig.ChainID,
-				InitialL1BaseFee:      arbostypes.DefaultInitialL1BaseFee,
-				ChainConfig:           chainConfig,
-				SerializedChainConfig: serializedChainConfig,
-			}
-			log.Warn("Created fake init message as L1Reader is disabled and serialized chain config from init message is not available", "json", string(serializedChainConfig))
+
+		// TODO: For the Task: Enabling execution only mode, getExecutionParsedInitMessage will be created, that will rely only on information available to Execution.
+		//
+		// TODO: Should this be part of getNewBlockchain()? Currently used by WriteOrTestGenblock() inside WriteOrTestBlockChain()
```

Comment:
Receiving the parsedInitMessage through an argument will make more clear that this getNewBlockchain should only be used when running an Execution node.

After the work that Piotr is doing, in this [task](https://linear.app/offchain-labs/issue/NIT-4201/consensus-calls-executionvalidate-to-check-parsedinitmessage), getNewBlockchain will use getExecutionParsedInitMsg.